smartclide-monitoring/src/main/java/eu/smartclide/contexthandling/services/GitRestCallService.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.InvalidPathException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+
+public class GitRestCallService {
+
+    private static final Logger logger = LoggerFactory.getLogger(GitRestCallService.class);
+
+    private final String accessTokenParam;
+    private final String baseUri = "https://gitlab.atb-bremen.de/api/v4/projects/";
+    private final String membershipParam = "&membership=true";
+    private final String paginationParam = "&per_page=100";
+    private final String refNameParam = "&ref_name=";
+    private final String sinceParam = "&since=2022-04-25T13:05:00"; // TODO change this based on requirement
+    private final String uriParams;
+    private final String uriPartForBranches = "/repository/branches/";
+    private final String uriPartForCommits = "/repository/commits/";
+    private final String uriPartForDiff = "/diff/";
+
+    public GitRestCallService(String accessTokenParam) {
+        this.accessTokenParam = "private_token=" + accessTokenParam;
+        this.uriParams = "?" + this.accessTokenParam + this.membershipParam + this.paginationParam;
+    }
+
+    /**
+     * generates gitMessages for given user
+     * creates separate message for the commit in all the branches
+     *
+     * @return List<GitMessage>
+     */
+    public List<GitMessage> getGitMessages() {
+        // first get all user projects
+        JsonArray projects = getUserProjects();
+        List<GitMessage> gitMessages = new LinkedList<>();
+        for (JsonElement project : projects) {
+            // get project id
+            String projectId = project.getAsJsonObject().get("id").getAsString();
+            // get all branches for given project, create a new GitMessage
+            JsonArray branches = getAllBranchesForGivenProject(projectId);
+            for (JsonElement branch : branches) {
+                String branchName = branch.getAsJsonObject().get("name").getAsString();
+                // get all commits for given branch and since
+                JsonArray commitsInBranch = getCommitsForGivenBranchAndSince(projectId, branchName, sinceParam);
+                for (JsonElement commit : commitsInBranch) {
+                    String commitId = commit.getAsJsonObject().get("id").getAsString();
+
+                    GitMessage gitMessage = new GitMessage();
+                    gitMessage.setHeader(GitMessageHeader.NEW_COMMIT.getHeader());
+                    gitMessage.setState("info");
+                    gitMessage.setUser(commit.getAsJsonObject().get("author_name").getAsString());
```

Comment:
It's probably a good idea to call `getAsJsonObject()` only once per for loop, and not every time you want to access one of the JsonElement's properties.